### PR TITLE
[OF-1262] feat: Add GaussianSplatComponent

### PR DESCRIPTION
**Pull request by MAG-AdamThorn**
A new component designed to allow for client applications to place and render Gaussian Splat assets within spaces.

### CODE REVIEW

File: Library/src/Multiplayer/Script/ComponentBinding/VolumetricModelSpaceComponentScriptInterface.cpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+DEFINE_SCRIPT_PROPERTY_STRING(VolumetricModelSpaceComponent, ExternalResourceAssetId);
+DEFINE_SCRIPT_PROPERTY_STRING(VolumetricModelSpaceComponent, ExternalResourceAssetCollectionId);
+
+DEFINE_SCRIPT_PROPERTY_VEC3(VolumetricModelSpaceComponent, Scale);
+DEFINE_SCRIPT_PROPERTY_VEC3(VolumetricModelSpaceComponent, Position);
+DEFINE_SCRIPT_PROPERTY_VEC4(VolumetricModelSpaceComponent, Rotation);
+
+DEFINE_SCRIPT_PROPERTY_TYPE(VolumetricModelSpaceComponent, bool, bool, IsVisible);
```
Comment (MAG-SamBirley) [inline Library/src/Multiplayer/Script/ComponentBinding/VolumetricModelSpaceComponentScriptInterface.cpp:42]: Should we add IsARVisible?

**Review by MAG-SamBirley — CHANGES_REQUESTED** [review]
Don't forget the new asset type!

File: Library/src/Multiplayer/Script/ComponentBinding/VolumetricModelSpaceComponentScriptInterface.cpp (hunk 1 above)
Comment (MAG-AdamThorn) [reply Library/src/Multiplayer/Script/ComponentBinding/VolumetricModelSpaceComponentScriptInterface.cpp]: Added.

File: Library/include/CSP/Multiplayer/Components/VolumetricModelSpaceComponent.h [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/// @file VolumetricModelSpaceComponent.h
+/// @brief Definitions and support for volumetric models.
```
[the remark this answers is not in the document]
Comment (MAG-AdamThorn) [reply Library/include/CSP/Multiplayer/Components/VolumetricModelSpaceComponent.h]: Agreed, I have added a little more info about Gaussian Splats here.

File: Library/include/CSP/Multiplayer/Components/GaussianSplatSpaceComponent.h [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+{
+	Name = 0,
+	ExternalResourceAssetId,
+	ExternalResourceAssetCollectionId,
+	Position,
+	Rotation,
+	Scale,
+	IsVisible,
+	IsARVisible,
+	ThirdPartyComponentRef,
```
Comment (MAG-SamBirley) [inline Library/include/CSP/Multiplayer/Components/GaussianSplatSpaceComponent.h:46]: What value does this have in the context of gsplats?

**Review by MAG-SamBirley — COMMENTED** [review]
Can we have a test to validate the gsplat component API?

File: Library/include/CSP/Multiplayer/Components/GaussianSplatSpaceComponent.h (hunk 3 above)
Comment (MAG-AdamThorn) [reply Library/include/CSP/Multiplayer/Components/GaussianSplatSpaceComponent.h:46]: What ThirdPartyComponentRef?

Comment (MAG-SamBirley) [reply Library/include/CSP/Multiplayer/Components/GaussianSplatSpaceComponent.h:46]: Yeah

**Comment by MAG-AdamThorn** [comment]
@MAG-SamBirley We discussed this when I originally implemented the Volumetric component. None of the other components currently have tests and in gsplat assets are massive and so if we elected to test the component in the future we would have to consider the best way of doing this.

**Comment by MAG-SamBirley** [comment]
> @MAG-SamBirley We discussed this when I originally implemented the Volumetric component. None of the other components currently have tests and in gsplat assets are massive and so if we elected to test the component in the future we would have to consider the best way of doing this.

Roger that 👍

File: Library/include/CSP/Multiplayer/Components/GaussianSplatSpaceComponent.h (hunk 3 above)
Comment (MAG-AdamThorn) [reply Library/include/CSP/Multiplayer/Components/GaussianSplatSpaceComponent.h]: Removed.

**Review by MAG-SamBirley — APPROVED** [review]

**Pull request merged**